File: app/src/main/java/net/tjado/usbgadget/DeviceInfoFragment.java
package net.tjado.usbgadget;

import android.graphics.Typeface;
import android.os.Bundle;
import android.text.Html;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.fragment.app.Fragment;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import java.io.BufferedReader;
import java.io.StringReader;
import java.util.HashMap;
import java.util.Map;
import java.util.TreeMap;

public class DeviceInfoFragment extends Fragment {

    private MutableLiveData<TreeMap<String,String>> deviceData;
    private View v;

    public DeviceInfoFragment() {}

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        this.v = inflater.inflate(R.layout.fragment_device_info, container, false);

        deviceData = new MutableLiveData<>();
        deviceData.setValue(new TreeMap<>(new DeviceInfoMapComparator()));

        deviceData.observe(getViewLifecycleOwner(), item -> {
            this.loadData(item);
        });

        GadgetShellApi gsa = new GadgetShellApi();
        gsa.updateDeviceInfo(deviceData);

        return v;
    }

    private void loadData(Map<String, String> deviceData) {

        LinearLayout list = this.v.findViewById(R.id.list_device_data);
        list.removeAllViews();

        View viHead = getLayoutInflater().inflate(R.layout.row_device_info, null);

        TextView tvHeadName = viHead.findViewById(R.id.name);
        tvHeadName.setText("Kernel Config Parameter");
        tvHeadName.setTypeface(null, Typeface.BOLD);

        TextView tvHeadValue = viHead.findViewById(R.id.value);
        tvHeadValue.setText("Value");
        tvHeadValue.setTypeface(null, Typeface.BOLD);
        list.addView(viHead);

        for (Map.Entry<String, String> entry : deviceData.entrySet()) {
            View vi = getLayoutInflater().inflate(R.layout.row_device_info, null);

            TextView tvName = vi.findViewById(R.id.name);
            tvName.setText(entry.getKey().toUpperCase());

            TextView tvValue = vi.findViewById(R.id.value);
            String value = entry.getValue();

            String color;
            switch (value) {
                case "y":
                    value = "Yes";
                    color = "#008000";
                    break;
                case "n":
                    value = "No";
                    color = "#ff0000";
                    break;
                case "NOT_SET":
                    value = "Not set";
                    color = "#ff0000";
                    break;
                default:
                    color = "#000000";
            }

            tvValue.setText(Html.fromHtml(String.format("<font color=%s>%s</font>", color, value), Html.FROM_HTML_MODE_LEGACY));

            list.addView(vi);
        }
    }
}

File: app/src/main/java/net/tjado/usbgadget/DeviceInfoFragment.kt
package net.tjado.usbgadget

import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.graphics.Typeface
import android.text.Html
import android.view.View
import androidx.fragment.app.Fragment
import androidx.lifecycle.MutableLiveData
import net.tjado.usbgadget.databinding.FragmentDeviceInfoBinding
import net.tjado.usbgadget.databinding.RowDeviceInfoBinding
import java.util.*


class DeviceInfoFragment : Fragment() {
    private var _binding: FragmentDeviceInfoBinding? = null
    private val binding get() = _binding!!

    private var deviceData: MutableLiveData<TreeMap<String, String>?>? = null

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentDeviceInfoBinding.inflate(inflater, container, false)
        val view = binding.root

        deviceData = MutableLiveData()
        deviceData!!.value = TreeMap(DeviceInfoMapComparator())
        deviceData!!.observe(viewLifecycleOwner) { item: TreeMap<String, String>? -> loadData(item) }
        val shellAPi = GadgetShellApi()
        shellAPi.updateDeviceInfo(deviceData!!)

        return view
    }

    private fun loadData(deviceData: Map<String, String>?) {
        binding.listDeviceData.removeAllViews()

        val rowHead = RowDeviceInfoBinding.inflate(layoutInflater)
        rowHead.name.text = "Kernel Config Parameter"
        rowHead.name.setTypeface(null, Typeface.BOLD)
        rowHead.value.text = "Value"
        rowHead.value.setTypeface(null, Typeface.BOLD)

        binding.listDeviceData.addView(rowHead.root)

        for (entry in deviceData!!.entries) {
            val rowEntry = RowDeviceInfoBinding.inflate(layoutInflater)
            rowEntry.name.text = entry.key.uppercase(Locale.getDefault())

            var value = entry.value
            var color: String
            when (value) {
                "y" -> {
                    value = "Yes"
                    color = "#008000"
                }
                "n" -> {
                    value = "No"
                    color = "#ff0000"
                }
                "NOT_SET" -> {
                    value = "Not set"
                    color = "#ff0000"
                }
                else -> color = "#000000"
            }
            rowEntry.value.text = Html.fromHtml(
                String.format("<font color=%s>%s</font>", color, value),
                Html.FROM_HTML_MODE_LEGACY
            )

            binding.listDeviceData.addView(rowEntry.root)
        }
    }
}